Osiris accept multiple schemas and exposes endpoint for logs

CermakM opened the pull request:
- It is now possible to specify optional url parameter to determine how
to process received data

Signed-off-by: Marek Cermak <macermak@redhat.com>

modified:   osiris/apis/build.py
modified:   osiris/schema/build.py
modified:   osiris/utils.py

Commit:
Osiris build endpoints accept kuberenetes event schema

- It is now possible to specify optional url parameter to determine how
to process received data

Signed-off-by: Marek Cermak <macermak@redhat.com>

modified:   osiris/apis/build.py
modified:   osiris/schema/build.py
modified:   osiris/utils.py

ghost commented:
Build failed.

- [thoth-coala ](https://zuul.thoth-station.ninja/logs/15/15/b8525004bd69ed5add1a0913cf863ebd21f74750/check/thoth-coala/989f76f/) : RETRY_LIMIT in 1m 47s

goern commented:
recheck

ghost commented:
Build failed.

- [thoth-coala ](https://zuul.thoth-station.ninja/logs/15/15/b8525004bd69ed5add1a0913cf863ebd21f74750/check/thoth-coala/5252402/) : FAILURE in 54s

Commit:
Introduce new endpoints specific to expected schema

- previous endpoints are preserved as default routes expecting
kubernetes event schema

Signed-off-by: Marek Cermak <macermak@redhat.com>

modified:   osiris/apis/build.py

ghost commented:
Build failed.

- [thoth-coala ](https://zuul.thoth-station.ninja/logs/15/15/a0e75e9b51572ea16dec236e126f01c9e43db92b/check/thoth-coala/0383eb2/) : FAILURE in 54s

Commit:
Allow to store logs separately from build information

- logs endpoint now implements `put` method to store logs for given
build (identified by build_id) independently of the build trigger. This
allows the Osiris API to run in separately from the observer and does not require
OpenShift credentials.

Signed-off-by: Marek Cermak <macermak@redhat.com>

modified:   osiris/aggregator.py
modified:   osiris/apis/build.py
modified:   osiris/schema/build.py

ghost commented:
Build failed.

- [thoth-coala ](https://zuul.thoth-station.ninja/logs/15/15/530030e750fb30c898f4e6c1652ef1157ab1e0b4/check/thoth-coala/b24666a/) : FAILURE in 57s

Commit:
Implement endpoint for ResourceInstance schema

- this endpoint is used for schema that is returned by Thoth OpenShift
API, as we want to move towards common interface, this schema is now
chosen to be the default

- add classmethods to BuildInfo and OCP to process ResourceInstance

- use common timestamp format (from thoth.common library)

Signed-off-by: Marek Cermak <macermak@redhat.com>

modified:   osiris/aggregator.py
modified:   osiris/apis/build.py
modified:   osiris/schema/build.py
modified:   osiris/schema/ocp.py

ghost commented:
Build failed.

- [thoth-coala ](https://zuul.thoth-station.ninja/logs/15/15/6b9da012b92868f98c9149b60b71e3068e9f4124/check/thoth-coala/40eceb4/) : FAILURE in 1m 00s

Commit:
Fix issues reported by coala

Signed-off-by: Marek Cermak <macermak@redhat.com>

modified:   osiris/apis/config.py
modified:   osiris/response.py
modified:   osiris/schema/base.py
modified:   osiris/schema/build.py
modified:   osiris/schema/config.py
modified:   osiris/schema/ocp.py

ghost commented:
Build succeeded.

- [thoth-coala ](https://zuul.thoth-station.ninja/logs/15/15/3be2e67ec07c6ecd82a8a907288789cc8fd95351/check/thoth-coala/25fc4d1/) : SUCCESS in 1m 11s

ghost commented:
Build succeeded (gate pipeline).

- [thoth-coala ](https://zuul.thoth-station.ninja/logs/15/15/3be2e67ec07c6ecd82a8a907288789cc8fd95351/gate/thoth-coala/e4f8c92/) : SUCCESS in 1m 22s
